rcl/test/rcl/test_subscription.cpp

```diff
@@ -130,56 +130,68 @@ TEST_F(CLASSNAME(TestSubscriptionFixture, RMW_IMPLEMENTATION), test_subscription
     rcl_ret_t ret = rcl_publisher_fini(&publisher, this->node_ptr);
     EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
   });
-  rcl_subscription_t subscription = rcl_get_zero_initialized_subscription();
+
   rcl_subscription_options_t subscription_options = rcl_subscription_get_default_options();
-  ret = rcl_subscription_init(&subscription, this->node_ptr, ts, topic, &subscription_options);
-  ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
-  OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT({
-    rcl_ret_t ret = rcl_subscription_fini(&subscription, this->node_ptr);
-    EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
-  });
-  EXPECT_EQ(strcmp(rcl_subscription_get_topic_name(&subscription), expected_topic), 0);
+
+  {
+    rcl_subscription_t subscription = rcl_get_zero_initialized_subscription();
+    ret = rcl_subscription_init(&subscription, this->node_ptr, ts, topic, &subscription_options);
+    ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+    OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT({
```

Comment:
I think we don't need to use this macro, and can just finalize the subscription at the end of this block. IMO it would be slightly more readable, but I'm okay leaving this as-is.

Reply:
@jacobperron Thank you for review. I rebased right now.

> I think we don't need to use this macro

Yeah, I remove this macro (easy to merge :sweat:)

Reply:
Mmm? Sorry, I made operation mistake.

We only need 32d03cd391d78588d4ea22ca7ba16f7dc2340735 at this PR, other commits([PR 
 564](https://github.com/ros2/rcl/pull/564)) are includes. Please wait I fix them

Reply:
I fixed, sorry for sloopy check.
And as obvious operation misstake, I used forced update.

Thank you

Reply:
No worries :) Thanks for changing it. See my latest comment regarding the other fini call.